src/snowflake/snowpark/functions.py

```diff
@@ -358,15 +360,19 @@ def lit(literal: LiteralType, _emit_ast: bool = True) -> Column:
         ---------------------------------------------------------------------------------------
         <BLANKLINE>
     """
+
+    assert not isinstance(
+        literal, Column
+    ), "Do not use lit(Column(...)), type hint does not allow this syntax."
```

Comment:
Is this just reinforcing the warning issued by the type hint? It makes sense to keep the assert if the warning is silent/can be easily ignored.
I see that in the rest of the file there's a lot of code of the form `n if isinstance(n, Column) else lit(n)` and this assertion is supposed to prevent folks from simply doing `lit(n)`?

Reply:
I decided to add an explicit assert here to prevent wrong usage. Unfortunately, in the code there's often something like `lit(Column(...))`. This is not correct though, because a `Column` can be an arbitrary expression. To fix this, I added the `isinstance(...)` code. I was debating to introduce a helper function, but felt there's not really a point for a single if stmt. Better to be explicit there.

Reply:
Makes sense, thank you for making this change!